provider/core/src/request.rs

```diff
@@ -244,48 +228,78 @@ impl Default for DataIdentifierCow<'_> {
 ///
 /// let locale1 = locale!("en-u-ca-buddhist");
 /// let data_locale = DataLocale::from(&locale1);
-/// let locale2 = data_locale.into_locale();
-///
-/// assert_eq!(locale1, locale2);
 /// ```
 ///
-/// If you are sure that you have no Unicode keywords, start with [`LanguageIdentifier`]:
-///
-/// ```
-/// use icu_locale_core::langid;
-/// use icu_provider::DataLocale;
-///
-/// let langid = langid!("es-CA-valencia");
-/// let data_locale = DataLocale::from(langid);
-/// let langid = data_locale.get_langid();
-///
-/// assert_eq!(langid, langid!("es-CA-valencia"));
-/// ```
-///
-/// [`DataLocale`] only supports `-u` keywords, to reflect the current state of CLDR data
+/// [`DataLocale`] only supports `-u-sd` keywords, to reflect the current state of CLDR data
 /// lookup and fallback. This may change in the future.
 ///
 /// ```
 /// use icu_locale_core::{locale, Locale};
 /// use icu_provider::DataLocale;
 ///
-/// let locale = "hi-t-en-h0-hybrid-u-attr-ca-buddhist"
+/// let locale = "hi-IN-t-en-h0-hybrid-u-attr-ca-buddhist-sd-inas"
 ///     .parse::<Locale>()
 ///     .unwrap();
-/// let data_locale = DataLocale::from(locale);
 ///
-/// assert_eq!(data_locale.into_locale(), locale!("hi-u-ca-buddhist"));
+/// assert_eq!(DataLocale::from(locale), DataLocale::from(locale!("hi-IN-u-sd-inas")));
 /// ```
-#[derive(PartialEq, Clone, Default, Eq, Hash)]
+#[derive(Clone, Default, Eq)]
 pub struct DataLocale {
```

Comment:
Suggestion: make it non exhaustive?

Reply:
I want it to be exhaustive with public fields, like `LanguageIdentifier` and `Locale`. We have a non-exhaustive `DataIdentifier`.